web/src/test/java/com/epam/rft/atsy/cucumber/candidate/CandidateCreationStepDefs.java

```diff
@@ -0,0 +1,207 @@
+
+  @Given("^other candidates data collected$")
+  public void other_candidates_data_collected() {
+    WebDriverWait webDriverWait = new WebDriverWait(getDriver(), TIMEOUT);
+
+    List<WebElement>
+        webElements =
+        webDriverWait
+            .until(visibilityOfAllElementsLocatedBy(By.cssSelector("tbody > tr[data-index]")));
+
+    candidateTableRows = new ArrayList<>();
+    for (WebElement element : webElements) {
+      List<WebElement> columns = element.findElements(By.tagName("td"));
+      String name = columns.get(0).getText();
+      String email = columns.get(1).getText();
+      String phone = columns.get(2).getText();
+      String position = columns.get(3).getText();
+
+      candidateTableRows.add(new CandidateTableRow(name, email, phone, position));
+    }
+  }
+
+  @Given("^the user is on the Candidate creation page$")
+  public void the_user_is_on_the_Candidate_creation_page() throws Throwable {
+    getDriver().get(CANDIDATE_CREATION_URL);
+  }
+
+  @Given("^the candidate details are empty$")
+  public void the_candidate_details_are_empty() throws Throwable {
+    WebElement descriptionElement = getDriver().findElement(By.id(DESCRIPTION_ID));
+    descriptionElement.clear();
+    assertTrue(descriptionElement.getText().isEmpty());
+  }
+
+  @Given("^the user enters name \"([^\"]*)\"$")
+  public void the_user_enters_name(String arg1) throws Throwable {
+    getDriver().findElement(By.id(NAME_ID)).sendKeys(arg1);
+  }
+
+  @Given("^the user enters e-mail address \"([^\"]*)\"$")
+  public void the_user_enters_e_mail_address(String arg1) throws Throwable {
+    getDriver().findElement(By.id(EMAIL_ID)).sendKeys(arg1);
+  }
+
+  @Given("^the user enters phone number \"([^\"]*)\"$")
+  public void the_user_enters_phone_number(String arg1) throws Throwable {
+    getDriver().findElement(By.id(PHONE_ID)).sendKeys(arg1);
+  }
+
+  @Given("^the user enters the place where the candidate has heard about the company \"([^\"]*)\"$")
+  public void the_user_enters_the_place_where_the_candidate_has_heard_about_the_company(String arg1)
+      throws Throwable {
+    getDriver().findElement(By.id(REFERER_ID)).sendKeys(arg1);
+  }
+
+  @Given("^the user enters the language level \"([^\"]*)\"$")
+  public void the_user_enters_the_language_level(String arg1) throws Throwable {
+    Select select = new Select(getDriver().findElement(By.id(LANGUAGE_SKILL_ID)));
+    select.selectByVisibleText(arg1);
+  }
+
+  @When("^the user clicks on the \"([^\"]*)\" button$")
+  public void the_user_clicks_on_the_button(String arg1) throws Throwable {
+    getDriver().findElement(By.id(SAVE_BUTTON_ID)).click();
+  }
+
+  @Then("^a \"([^\"]*)\" message is shown under the \"([^\"]*)\" field$")
+  public void a_message_is_shown_under_the_field(String msg, String field) throws Throwable {
+    String nameOrPhoneErrorSelector = ".list-unstyled > li:nth-child(1)";
+    String phoneErrorSelector = "#phoneDiv > div > div > ul > li:nth-child(1)";
```

Comment:
Please avoid the use of too complex locators. If it is possible, ask the ui developers for id.